pkg/apis/kubermatic/v1/settings.go

```diff
@@ -129,9 +129,27 @@ type SettingSpec struct {
 	// StaticLabels are a list of labels that can be used for the clusters.
 	StaticLabels []StaticLabel `json:"staticLabels,omitempty"`
 
+	// Annotations are the settings for the annotations in KKP UI.
+	Annotations AnnotationSettings `json:"annotations,omitempty"`
+
 	// TODO: Datacenters, presets, user management, Google Analytics and default addons.
 }
 
+// AnnotationSettings is the settings for the annotations.
+type AnnotationSettings struct {
+	// +kubebuilder:default:={"kubectl.kubernetes.io/last-applied-configuration", "kubermatic.io/initial-application-installations-request", "kubermatic.io/initial-machinedeployment-request"}
```

Comment:
So when I do not configure hiddenAnnotations, will Kubernetes now on the initial CREATE request default this field to this list of annotations? I.e. just once? What happens if we change this default in the future, will it affect existing KKP setups?

Personally I do not like these super complex default values inside the CRD and would rather keep it nil and default it at runtime. Setting the current random 3 annotations as the default I think will soon cause issues when we want to add another one and suddenly have to migrate KubermaticSettings.

Reply:
I don't agree with your points here. If given an option, I'll always prefer defaulting using Kubebuilder markers since they get documented into the CRD automatically, and no special code is required for it. These annotations are not "random"; we discussed and settled for these annotations.

Also, I don't understand how an array of strings as default is "super complex"?

Anyway, we don't have mutation/validation webhooks for KubermaticSettings, and I don't think we need them here as well since I can't see any solid case of migration. For 2.26, most of the annotations will be added before release. In case if we need something in the future afterwards, we can either introduce either mutation webhooks for just this one use case or rely on KKP API for this, like we already do for [creation](https://github.com/kubermatic/dashboard/blob/main/modules/api/pkg/provider/kubernetes/settings.go#L69-L107) of `globalsettings` which is the default.